src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -73,25 +77,50 @@ protected ItemUseHandler getItemUseHandler() {
         return e -> {
             Player p = e.getPlayer();
             ItemStack item = e.getItem();
+            ItemMeta meta = item.getItemMeta();
             e.cancel();
 
-            int index = selectedMode.getOrDefault(p.getUniqueId(), 0);
+            int index = 0;
+            if (PersistentDataAPI.hasString(meta, key)) {
+                for (int i = 0; i < modes.size(); i++) {
+                    if (modes.get(i).getItemId().equals(PersistentDataAPI.getString(meta, key, modes.get(index).getItemId()))) {
+                        index = i;
+                        break;
+                    }
+                }
+            }
```

Comment:
Note: Don't forget the imports.
For loops are possibly substitutable via `forEach`, `Stream`s. Although in this case the only way to replace it is via a Stream.
```suggestion
            int index = !PersistentDataAPI.hasString(meta, key) ? 0 :
                IntStream.range(0, modes.size()).
                filter(i -> modes.get(i).getItemId().equals(
                    PersistentDataAPI.getString(meta, key, modes.get(0).getItemId())
                )).findFirst()
                .orElse(0);
```